pkg/logs/input/windowsevent/tailer.go

```diff
@@ -68,10 +66,150 @@ func (t *Tailer) toMessage(event string) (*message.Message, error) {
 	if err != nil {
 		return &message.Message{}, err
 	}
+	if strings.HasPrefix(t.config.ChannelPath, "Microsoft-ServiceFabric/") {
+		mapTaskIDToTaskName(mv)
+	}
 	jsonEvent, err := mv.Json(false)
 	if err != nil {
 		return &message.Message{}, err
 	}
 	log.Debug("Sending JSON: ", string(jsonEvent))
 	return message.NewMessage(jsonEvent, message.NewOrigin(t.source), message.StatusInfo), nil
 }
+
+// mapTaskIDToTaskName looks for the TASK_ID in {"Event": {"System": {"Task": <TASK_ID> }}}
+// and maps it to the name of the task that match in the Microsoft Task Codes
+func mapTaskIDToTaskName(mv mxj.Map) {
+	if evt, ok := mv["Event"]; ok {
+		if sys, ok := evt.(map[string]interface{})["System"]; ok {
```

Comment:
Shall we create a struct to avoid doing all those cast here ?

Reply:
Not sure we can because we might not have the full tree everytime, that's why there is a lot of presence checking

Reply:
Not relevant anymore, I found a better way of traversing the object thanks to the lib